mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/options/AuthenticationMechanismType.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.mongodb.options;
+
+/**
+ * Available authentication methods for MongoDB.
+ *
+ * @see com.mongodb.MongoCredential
+ * @author Davide D'Alto
+ */
+public enum AuthenticationMechanismType {
+
+	GSSAPI("GSSAPI"),
+	MONGODB_CR("MONGODB-CR"),
```

Comment:
> so it has to be MONGO-CR

Why that, we don't pass that value to MongoDB as is, right?

> note that MONGODB_CR will work anyway.

Ah, I just see you do some manual parsing above. That can be left to the configuration parser to deal with (provided the toString values of the enum members are used):

    AuthenticationMechanismType authenticationMechanism = propertyReader
        .property( MongoDBProperties.AUTHENTICATION_MECHANISM, AuthenticationMechanismType.class )
        .withDefault( AuthenticationMechanismType.MONGODB_CR )
        .getValue();

That way the value can also be given as an actual enum member when programmatically bootstrapping OGM. That was actually the main motivation for bringing up the enum topic, otherwise we could stick to the current implementation.